meson.build

```diff
@@ -0,0 +1,81 @@
+project('libzseek', 'c',
+    default_options: ['c_std=c11', 'warning_level=3', 'werror=true'],
+    license: 'BSD-3-Clause',
+    version: '3.2.0')
+add_project_arguments('-D_GNU_SOURCE', language: 'c')
+
+
+threads_dep = dependency('threads')
+zstd_dep = dependency('libzstd', version: '>=1.4.9')
+lz4_dep = dependency('liblz4', version: '>=1.8.3')
+check_dep = dependency('check')
+# From https://mesonbuild.com/howtox.html#add-math-library-lm-portably
+cc = meson.get_compiler('c')
+m_dep = cc.find_library('m', required : false)
+
+libzseek = library('zseek',
+    'src/buffer.c',
+    'src/cache.c',
+    'src/common.c',
+    'src/compress.c',
+    'src/decompress.c',
+    'src/seek_table.c',
+    dependencies: [threads_dep, zstd_dep, lz4_dep],
+    gnu_symbol_visibility: 'hidden',
+    install: true,
+    soversion: '3.2.0')
+libzseek_dep = declare_dependency(
+    include_directories: 'src',
+    link_with: libzseek)
+
+
+libzseek_example = executable('libzseek_example',
+    'test/example.c',
+    dependencies: libzseek_dep)
+libzseek_benchmark = executable('libzseek_benchmark',
+    'test/benchmark.c',
+    dependencies: [libzseek_dep, m_dep])
+
+# Extract object file to use directly un-exported symbols. See
+# https://mesonbuild.com/Build-targets.html#object-files
+cache_o = libzseek.extract_objects('src/cache.c')
+test_cache = executable('test_cache',
+    'test/test_cache.c',
+    dependencies: [check_dep],
+    objects: [cache_o])
+test('test_cache', test_cache)
+
+buffer_o = libzseek.extract_objects('src/buffer.c')
+test_buffer = executable('test_buffer',
+    'test/test_buffer.c',
+    dependencies: [check_dep],
+    objects: [buffer_o])
+test('test_buffer', test_buffer)
+
+
+install_headers('src/zseek.h')
+
+project_desc = 'Random access decompression library'
+pkg_mod = import('pkgconfig')
+pkg_mod.generate(libzseek,
+    description: project_desc,
+    filebase: 'libzseek',
+    name: meson.project_name(),
+    url: 'https://github.com/foxeng/libzseek')
+
+
+doxyfile = configure_file(input: 'Doxyfile.in',
+    output: 'Doxyfile',
+    configuration: {
+        'NAME': meson.project_name(),
+        'DESCRIPTION': project_desc,
+        'VERSION': meson.project_version(),
+        'TOP_SRCDIR': meson.project_source_root(),
+        'TOP_BUILDDIR': meson.project_build_root(),
+    },
+    install: false)
```

Comment:
`install: false` is the default and you do not need to specify it. If you do, that raises the minimum version of Meson to at least 0.50, independent of the previous `project_*_root()` usage.

Reply:
Had noted it's the default, but included it nonetheless since from the docs:

| When omitted it defaults to true when install_dir is set and not empty, false otherwise.

On a second thought though, that's almost irrational guarding against a future self, so will remove it and fallback to the default, thanks!